src/probe/probe.c

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2019 Intel Corporation. All rights reserved.
+//
+// Author: Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+
+#include <sof/probe/probe.h>
+#include <sof/trace/trace.h>
+#include <user/trace.h>
+
+#define trace_probe(__e, ...) \
+	trace_event(TRACE_CLASS_PROBE, __e, ##__VA_ARGS__)
+#define tracev_probe(__e, ...) \
+	tracev_event(TRACE_CLASS_PROBE, __e, ##__VA_ARGS__)
+#define trace_probe_error(__e, ...) \
+	trace_error(TRACE_CLASS_PROBE, __e, ##__VA_ARGS__)
+
+int probe_init(struct probe_dma *probe_dma)
+{
+	return 0;
```

Comment:
Why are all these functions empty, should they be in a new PR ?

Reply:
Implementation was done in PR 1852, but review was taking a long time, so to aid this process, and provide proper PR base for probe application, we've decided to split this. It will also separate interface and implementation discussions, which would hopefully allow for faster and better review process.

Reply:
agree, it's much better to look at the interface first.